src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -250,6 +271,82 @@ public Mono<VposAuthResponseDto> requestCreditCardAuthorization(AuthorizationReq
                 );
     }
 
+    public Mono<Tuple2<String, FieldsDto>> requestNpgBuildSession(AuthorizationRequestData authorizationData) {
+        return uniqueIdUtils.generateUniqueId()
+                .flatMap(
+                        orderId -> {
+                            UUID correlationId = UUID.randomUUID();
+                            URI returnUrlBasePath = URI.create(npgSessionUrlConfig.basePath());
+                            URI outcomeResultUrl = UriComponentsBuilder.fromUriString(
+                                    returnUrlBasePath.resolve(npgSessionUrlConfig.outcomeSuffix()).toString()
+                                            .concat("&clientId=IO&transactionId=")
+                                            .concat(authorizationData.transactionId().value())
+                            ).build().toUri();
+                            URI merchantUrl = returnUrlBasePath;
+                            URI cancelUrl = returnUrlBasePath.resolve(npgSessionUrlConfig.cancelSuffix());
+                            URI notificationUrl = UriComponentsBuilder
+                                    .fromHttpUrl(npgSessionUrlConfig.notificationUrl())
+                                    .build(
+                                            Map.of(
+                                                    "orderId",
+                                                    orderId,
+                                                    "paymentMethodId",
+                                                    authorizationData.paymentInstrumentId()
+                                            )
+                                    );
+
+                            return npgClient.buildForm(
+                                    correlationId,
+                                    merchantUrl,
+                                    outcomeResultUrl,
+                                    notificationUrl,
+                                    cancelUrl,
+                                    orderId,
+                                    null,
+                                    NpgClient.PaymentMethod.fromServiceName(authorizationData.paymentMethodName()),
+                                    npgDefaultApiKey,
+                                    authorizationData.contractId().get()
```

Comment:
use `orElseThrow` method to throw an appropriate exception here in case of missing contractId at runtime

Reply:
9f28de4c81b19ccd10d80043d596998fe86cc6dd